bloom-housing/bloom
# feat: pwdless sign in flow

# Pull Request Template

## Issue Overview

This PR addresses #3818 


## Description

Ties the frontend and backend together for the passwordless sign in flow. This doesn't cover the create account flow yet.

## How Can This Be Tested/Reviewed?

You will need to toggle on SHOW_PWDLESS=TRUE on the public site.

Review the sign-in flow in the [Figma](https://www.figma.com/file/2LSD4B4yBIxAUGbZQCVOCv/Passwordless?type=design&node-id=1607-68638&mode=design&t=K1pMXNPSCXutTzRy-0).

Known differences:
* The error messages include the email
* Some spacing is in UIC and differs from the designs, namely the letter spacing on notes on fields and some of the padding around fields. There are also some spacing differences internally to the card, but those were ok-d by design since they differ from what is in Seeds & elsewhere in the app.

You should be able to toggle between the password and code versions on the sign in page. A code should send successfully and the email should match the design. Entering an incorrect code should decrease the login tries remaining and throw an error. You should be able to resend the code. Clicking the use password instead button should redirect you to the sign in page with the password fieldset toggled on. Mandated accounts should also still work, so if you visit a listing logged out and apply --> sign in with a code, you should be redirected back to the listing after signing in.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
